### PR TITLE
Add close button to dismiss builds from overview

### LINKED ISSUE #3327: "a build has completed" cannot be dismissed
screenshot from 2015-06-18 19 24 06

There's no "x" button... is this intended?  Do these ever go away?

screenshot from 2015-06-18 19 25 37

### LINKED ISSUE #3416: Completed builds on overview disappear immediately when clicked
If you wait exactly a minute after a build completes on the overview and click it to get details, it disappears immediately. See discussion in #3344 and #3349 for details.

### PR DESCRIPTION
**Pull request by spadgett**
Fixes #3327
Fixes #3416

openshift_web_console

### CODE REVIEW

File: assets/app/scripts/directives/resources.js [hunk 1]
```diff
@@ -51,11 +51,24 @@ angular.module('openshiftConsole')
     };
   })
   .directive('triggers', function() {
+    var hideBuildKey = function(build) {
+      return 'hide-' + build.metadata.name;
```
Comment (liggitt) [inline assets/app/scripts/directives/resources.js]: needs namespace as well

**Comment by liggitt** [comment]
LGTM otherwise

**Comment by spadgett** [comment]
@liggitt another look?

**Comment by liggitt** [comment]
What does it look like when the build text is long (or the screen is narrow) and it needs to wrap?

**Comment by spadgett** [comment]
screen shot 2015-07-07 at 12 49 28 pm

**Comment by spadgett** [comment]
@liggitt Are you OK with showing the close button while the build is still pending or running?

**Comment by liggitt** [comment]
Wrapping looks good. Probably shouldn't show the close button while pending/running

File: assets/app/scripts/directives/resources.js [hunk 2]
```diff
@@ -51,11 +51,24 @@ angular.module('openshiftConsole')
     };
   })
   .directive('triggers', function() {
+    var hideBuildKey = function(build) {
+      return 'hide-' + build.metadata.namespace + '/' + build.metadata.name;
```
Comment (liggitt) [inline assets/app/scripts/directives/resources.js]: nit, maybe use a character that can't appear in a namespace, like `hide/`?

**Comment by spadgett** [comment]
@liggitt mind if I change complete builds to display for 5 minutes now that you can dismiss them?

**Comment by liggitt** [comment]
that feels long to me, but I don't feel strongly

**Comment by spadgett** [comment]
Removed close button for running builds and updated name of the sessionStorage key.

**Comment by liggitt** [comment]
when the page is wide, does the `X` feel really disconnected from the build name? We don't have a visible container around the build... wondering if a "Hide" link following the build text would be more obviously connected. That might also work better if we start adding other actions like "Rebuild" or "View logs"

**Comment by spadgett** [comment]
Your call.

screen shot 2015-07-07 at 3 12 46 pm

**Comment by liggitt** [comment]
That's pretty detached. Does the link look reasonable?

**Comment by spadgett** [comment]
openshift_web_console

**Comment by liggitt** [comment]
Spacing, trailing `.`, and wording of `Hide` might need tweaking, but I think that's probably better overall... gives is a reasonable place to hang other actions, doesn't get confused with the red X of a failed build, etc

**Comment by spadgett** [comment]
Hide, Dismiss, Close, Remove, Done...?

screen shot 2015-07-07 at 3 42 31 pm

**Comment by spadgett** [comment]
Snuck in a fix for #3416 while in the code.

**Comment by liggitt** [comment]
heh, I meant the trailing period on the "A build of ___ failed." looked odd with the "Hide" link so closely spaced... with more space, the existing period looks fine.

"Dismiss" was what came to mind first...

**Comment by spadgett** [comment]
screen shot 2015-07-07 at 3 59 24 pm

**Comment by liggitt** [comment]
LGTM, [merge]

**Pull request merged**